clang/lib/Serialization/ASTReaderDecl.cpp

```diff
@@ -1016,21 +1003,20 @@ void ASTDeclReader::VisitFunctionDecl(FunctionDecl *FD) {
   }
   case FunctionDecl::TK_DependentFunctionTemplateSpecialization: {
     // Templates.
-    UnresolvedSet<8> TemplDecls;
-    unsigned NumTemplates = Record.readInt();
-    while (NumTemplates--)
-      TemplDecls.addDecl(readDeclAs<NamedDecl>());
+    UnresolvedSet<8> Candidates;
+    unsigned NumCandidates = Record.readInt();
```

Comment:
It looks like this should be `uint64_t`, although it does not seem to do what the "label says" it does.

Reply:
@shafik  The "convention" here seems to be to use the type of the member being deserialized (see lines 1082 & 1122). The `UnresolvedSet` is only used to pass the candidates to `DependentFunctionTemplateSpecializationInfo::Create`; the actual candidates are stored as trailing objects & the number of candidates is stored in `DependentFunctionTemplateSpecializationInfo::NumCandidates` (which has the type `unsigned`).